index.js

```diff
@@ -163,9 +163,15 @@ module.exports = fp(function (fastify, options, next) {
     return cipher.toString('base64') + ';' + nonce.toString('base64')
   })
 
-  fastify
-    .register(require('@fastify/cookie'))
-    .register(fp(addHooks))
+  if (fastify.hasDecorator('parseCookie')) {
+    // `@fastify/cookie` might already be registered.
+    fastify
+      .register(fp(addHooks))
+  } else {
+    fastify
+      .register(require('@fastify/cookie'))
+      .register(fp(addHooks))
+  }
```

Comment:
@jsumners 
Thanks for your comment. If the `.hasPlugin` method is acceptable to be added, I can send PR.
Maybe I will add the method around these lines: https://github.com/fastify/fastify/blob/b6b755339cfbc0f63c5e06559a69d44d49a369a7/fastify.js#L318-L331

Reply:
+1!

Reply:
Ok! I'll create another PR to the `fastify` repository.

Reply:
This has shipped in `fastify@4`.

Reply:
@jsumners 
Thanks! I'll update this PR.

Reply:
@jsumners 
I updated it to use `hasPlugin` method. I also bumped `@fastify/cookie` version in `dependencies`. This is because `@fastify/cookie`'s name has been changed in `v7.0.0`. 
https://github.com/fastify/fastify-cookie/compare/v6.0.0...v7.0.0#diff-8a54c049b81d2642e951ea697e44aca5d74fb1b990552e8dd0383fe51917da95R94-R95
Could you review it again, please?

Reply:
I wonder if we should check `fastify-cookie` too...
What do you think about it?

Reply:
>  What do you think about it?

No, we should not support deprecated plugin in latest version.
But, if it is back-ported to `fastify@3`, then it should be `fastify-cookie`.

Reply:
@climba03003 
Thanks for your comment. I agree with you.